### PR TITLE
Add (simple) impl on (bool ^ int)

### PR DESCRIPTION
**Pull request by mcfongtw**
Utilize bit-wise XOR operation (in Java) to perform (bool: 1/0) ^ int

Tested result:
$ voc -v bool_xor_int.py 
Compiling bool_xor_int.py ...
Writing ./python/bool_xor_int.class ...
$ java -classpath ../voc/dist/python-java-support.jar:. python.bool_xor_int
bool ^ int
0
1
3
1
0
2
$ cat bool_xor_int.py 
print('bool ^ int')
print(True ^ 1)
print(True ^ 0)
print(True ^ 2)
print(False ^ 1)
print(False ^ 0)
print(False ^ 2)

### CODE REVIEW

File: tests/datatypes/test_bool.py [hunk 1]
```diff
@@ -32,6 +32,13 @@ class UnaryBoolOperationTests(UnaryOperationTestCase, TranspileTestCase):
 class BinaryBoolOperationTests(BinaryOperationTestCase, TranspileTestCase):
     data_type = 'bool'
 
+    def test_xor_int(self):
+        self.assertCodeExecution("""
+            x = True
+            x ^ 2
+            #expect 3
+            """)
+
```
Comment (freakboy3742) [inline tests/datatypes/test_bool.py]: This extra test is not necessary; the test for xor_int is automatically generated, the `not_implemented` line that you delete below just tell the test suite that the test is now expected to pass.

**Comment by mcfongtw** [comment]
No idea on the possible cause of CI failure and also tried to run locally, and it seemed to pass though.

$ python3 setup.py test -s tests.builtins.test_list.BuiltinListFunctionTests
running test
running egg_info
writing voc.egg-info/PKG-INFO
writing top-level names to voc.egg-info/top_level.txt
writing dependency_links to voc.egg-info/dependency_links.txt
writing entry points to voc.egg-info/entry_points.txt
reading manifest file 'voc.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'Makefile'
warning: no files found matching 'requirements*.txt'
writing manifest file 'voc.egg-info/SOURCES.txt'
running build_ext
test_None (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_NotImplemented (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_bool (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_bytearray (tests.builtins.test_list.BuiltinListFunctionTests) ... expected failure
test_bytes (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_class (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_complex (tests.builtins.test_list.BuiltinListFunctionTests) ... ok

>> test_dict (tests.builtins.test_list.BuiltinListFunctionTests) ... expected failure

test_float (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_frozenset (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_int (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_list (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_range (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_set (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_slice (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_str (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_tuple (tests.builtins.test_list.BuiltinListFunctionTests) ... ok

----------------------------------------------------------------------
Ran 17 tests in 9.121s

OK (expected failures=2)

**Comment by mcfongtw** [comment]
Have been able to reproduce the failure locally, which happened at a test case that belongs to not_implemented section... ?

$ python3 setup.py test -s tests.builtins.test_list.BuiltinListFunctionTests
.........
test_None (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_NotImplemented (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_bool (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_bytearray (tests.builtins.test_list.BuiltinListFunctionTests) ... expected failure
test_bytes (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_class (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_complex (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
_**test_dict (tests.builtins.test_list.BuiltinListFunctionTests) ... unexpected success**_
test_float (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_frozenset (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_int (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_list (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_range (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_set (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_slice (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_str (tests.builtins.test_list.BuiltinListFunctionTests) ... ok
test_tuple (tests.builtins.test_list.BuiltinListFunctionTests) ... ok

----------------------------------------------------------------------
Ran 17 tests in 7.317s

FAILED (expected failures=1, unexpected successes=1)

**Comment by eliasdorneles** [comment]
Thank you @mcfongtw !

**Pull request merged**